setup.py

```diff
@@ -31,31 +31,31 @@ def find_version(*file_paths):
 
 install_requires = [
     'cached-property >= 1.2.0, < 2',
-    'docopt >= 0.6.1, < 0.7',
-    'PyYAML >= 3.10, < 4.3',
-    'requests >= 2.6.1, != 2.11.0, != 2.12.2, != 2.18.0, < 2.23',
-    'texttable >= 0.9.0, < 0.10',
-    'websocket-client >= 0.32.0, < 1.0',
-    'docker[ssh] >= 3.7.0, < 4.0.2',
-    'dockerpty >= 0.4.1, < 0.5',
+    'docopt >= 0.6.1, < 1',
+    'PyYAML >= 3.10, < 5',
+    'requests >= 2.20.0, < 3',
+    'texttable >= 0.9.0, < 1',
+    'websocket-client >= 0.32.0, < 1',
+    'docker[ssh] >= 3.7.0, < 5',
+    'dockerpty >= 0.4.1, < 1',
     'six >= 1.3.0, < 2',
     'jsonschema >= 2.5.1, < 3',
 ]
 
 
 tests_require = [
-    'pytest',
+    'pytest < 6',
 ]
 
 
 if sys.version_info[:2] < (3, 4):
-    tests_require.append('mock >= 1.0.1')
+    tests_require.append('mock >= 1.0.1, < 2')
```

Comment:
That would downgrade people with mock 2.0.0 and upstream is at 3.0.5 by now, so that change makes things worse.

Reply:
The idea here was just to use the current minimal boundaries to set the upper ones as it's a guaranty (as a safe move).
Then we should to bump every requirement independently (including `requirements.txt` so that the binary can evolve too).

Reply:
> The idea here was just to use the current minimal boundaries to set the upper ones as it's a guaranty (as a safe move).

Original `mock >= 1.0.1` pulled in 3.0.5 effectively so it should have been `mock <4` not `mock <2` to allow the same version as before.

> Then we should to bump every requirement independently (including `requirements.txt` so that the binary can evolve too).

That sounds right for texttable and pyyaml but not for mock.
Is there a guarantee that this branch is not ending up in a release? Is there a ticked or pull request about fixing the mock dependency regression now?

Reply:
Sure! I'll do the respective pull requests today :smiley: